tests/validation_tests/round_four_specific_validations/test_tf_r4_specific_validations.py

```diff
@@ -803,3 +805,59 @@ def test_validate_funding_profiles_funding_secured_not_null():
             row_indexes=[49],
         )
     ]
+
+
+def validate_test_postcodes():
+    project_detail_df = pd.DataFrame(
+        index=[7, 8, 9, 10, 11, 12, 13],
+        data=[
+            # Single postcode
+            {
+                "Locations": " AB1 2CD ",
+                "Postcodes": "AB1 2CD",
+            },
+            # Comma seperated list
+            {
+                "Locations": "EF3 4GH IJ5 6KL KL7 8MN",
+                "Postcodes": "EF3 4GH, IJ5 6KL, KL7 8MN",
+            },
+            # Space seperated list
+            {
+                "Locations": "EF3 4GH IJ5 6KL KL7 8MN",
+                "Postcodes": "EF3 4GH IJ5 6KL KL7 8MN",
+            },
+            # Dash seperated list
+            {
+                "Locations": "EF3 4GH IJ5 6KL KL7 8MN",
+                "Postcodes": "EF3 4GH - IJ5 6KL - KL7 8MN",
+            },
+            # Blank cell - will not be picked up
+            {
+                "Locations": pd.NA,
+                "Postcodes": "",
+            },
+            # Not postcode format
+            {
+                "Locations": "NOT A POSTCODE",
+                "Postcodes": "NOT A POSTCODE",
+            },
+            # Incorrect postcode - starts with a number
+            {
+                "Locations": "12A 2CD",
+                "Postcodes": "12A 2CD",
+            },
+        ],
+    )
+    workbook = {"Project Details": project_detail_df}
+
+    failures = validate_postcodes(workbook)
+
+    assert failures == [
+        TownsFundRoundFourValidationFailure(
+            sheet="Project Details",
+            section="Project Details",
+            column="Postcodes",
+            message="You entered an incorrect postcode. You must enter a postcode like SW1A 2AA",
+            row_indexes=[12, 13],
+        )
+    ]
```

Comment:
Could a case be added for when Locations has a value but Postcodes is an empty string